lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -607,79 +610,110 @@ Ufe::Attribute::Type usdTypeToUfe(const SdfValueTypeName& usdType)
-                { PXR_NS::SdrPropertyTypes->Vector, PXR_NS::SdfValueTypeNames->Vector3f },
-                { PXR_NS::SdrPropertyTypes->Matrix, PXR_NS::SdfValueTypeNames->Matrix4d } };
+                { SdrPropertyTypes->Point, SdfValueTypeNames->Point3f },
+                { SdrPropertyTypes->Normal, SdfValueTypeNames->Normal3f },
+                { SdrPropertyTypes->Vector, SdfValueTypeNames->Vector3f },
+                { SdrPropertyTypes->Matrix, SdfValueTypeNames->Matrix4d } };
         TokenToSdfTypeMap::const_iterator it
             = tokenTypeToSdfType.find(shaderProperty->GetTypeAsSdfType().second);
         if (it != tokenTypeToSdfType.end()) {
-            Ufe::Attribute::Type type = usdTypeToUfe(it->second);
-            if (type == Ufe::Attribute::kString && !shaderProperty->GetOptions().empty()) {
-                return Ufe::Attribute::kEnumString;
-            } else {
-                return type;
-            }
+            retVal = _UsdTypeToUfe(it->second);
         } else {
 #if PXR_VERSION < 2205
             // Pre-22.05 boolean inputs are special:
             if (shaderProperty->GetType() == SdfValueTypeNames->Bool.GetAsToken()) {
-                return usdTypeToUfe(PXR_NS::SdfValueTypeNames->Bool);
-            }
+                retVal = _UsdTypeToUfe(SdfValueTypeNames->Bool);
+            } else
 #endif
-            // There is no Matrix3d type in Sdr, so we need to infer it from Sdf until a fix similar
-            // to what was done to booleans is submitted to USD. This also means that there will be
-            // no default value for that type.
-            if (shaderProperty->GetType() == SdfValueTypeNames->Matrix3d.GetAsToken()) {
-                return usdTypeToUfe(PXR_NS::SdfValueTypeNames->Matrix3d);
+                // There is no Matrix3d type in Sdr, so we need to infer it from Sdf until a fix
+                // similar to what was done to booleans is submitted to USD. This also means that
+                // there will be no default value for that type.
+                if (shaderProperty->GetType() == SdfValueTypeNames->Matrix3d.GetAsToken()) {
+                retVal = _UsdTypeToUfe(SdfValueTypeNames->Matrix3d);
+            } else {
+                retVal = Ufe::Attribute::kGeneric;
             }
-            return usdTypeToUfe(PXR_NS::SdfValueTypeNames->Token);
         }
     } else {
-        Ufe::Attribute::Type type = usdTypeToUfe(typeName);
-        if (type == Ufe::Attribute::kString && !shaderProperty->GetOptions().empty()) {
-            return Ufe::Attribute::kEnumString;
-        } else {
-            return type;
+        retVal = _UsdTypeToUfe(typeName);
+    }
+
+    if (retVal == Ufe::Attribute::kString && !shaderProperty->GetOptions().empty()) {
+        retVal = Ufe::Attribute::kEnumString;
+    }
+
+    return retVal;
+}
+
+Ufe::Attribute::Type usdTypeToUfe(const PXR_NS::UsdAttribute& usdAttr)
```

Comment:
New function taking a full USD attribute. This will allow picking the right type.